Parser/pegen/pegen.c

```diff
@@ -2074,3 +2074,43 @@ _PyPegen_make_module(Parser *p, asdl_seq *a) {
     }
     return Module(a, type_ignores, p->arena);
 }
+
+// Error reporting helpers
+
+expr_ty
+_PyPegen_get_invalid_target(expr_ty e)
+{
+    if (e == NULL) {
+        return NULL;
+    }
+
+#define VISIT_CONTAINER(CONTAINER, TYPE) do { \
+        Py_ssize_t len = asdl_seq_LEN(CONTAINER->v.TYPE.elts);\
+        for (Py_ssize_t i = 0; i < len; i++) {\
+            expr_ty other = asdl_seq_GET(CONTAINER->v.TYPE.elts, i);\
+            expr_ty child = _PyPegen_get_invalid_target(other);\
+            if (child != NULL) {\
+                return child;\
+            }\
+        }\
+    } while (0)
+
+    switch (e->kind) {
```

Comment:
Could you maybe add a comment explaining why we only need `List_kind` and `Tuple_kind` here? It's because only they are valid assignment targets, but can contain invalid ones. I can easily see myself in the future wondering why this is.

Reply:
> Could you maybe add a comment explaining why we only need `List_kind` and `Tuple_kind` here?

There are others like `Starred_kind`, `Name_kind`,....

For `List_kind` and `Tuple_kind` those are handled like that because they can contain valid assignment targets recursively but for instance `Starred_kind` is also handled (not recursively) differently to get the value (so the error is the same as the old parser).

Reply:
Yes, I'm only talking about containers. I see `VISIT_CONTAINER` and I'm instantly thinking where aren't all containers handled here, like `Set_kind` or `Dict_kind`? And that's because the whole set is the error target, unlike lists and tuples, where the invalid target is one of the elements. It's probably a bit over-explaining things, but it wouldn't hurt IMHO.

Reply:
> but it wouldn't hurt IMHO.

I see what you mean now and I agree is good to explain it there :+1: